### ISSUE TITLE
Enhance Usage Documentation

### ISSUE THREAD

**Issue by niccokunzmann** [collective/icalendar#626]
When I skip over [usage.html], I think, this can be structured in a better way, more descriptive in places and also have more content.

Here is a list of what I think can be improved. For each item, please open a separate pull request.

- [x] Change all instances of `.. code:: pycon` to `..  code-block:: pycon`.
  - [x] Fix any broken doctests as a result of the previous change.
- [ ] change RFC 2445 to 5545 where you can find it
- [ ] File structure and overview can be merged.
- [ ] SUMMARY: might not be legal in the RFC, check it and make it `X-CAL-DESCRIPTION` or something used eitherway
- [ ] How to open a file with a `Calendar` object
- [ ] How to save a file
- [x] How to add an event https://icalendar.readthedocs.io/en/stable/how-to/attendees.html and https://icalendar.readthedocs.io/en/stable/how-to/usage.html#subcomponents
- [x] How to add an event in the local time zone using `tzlocal` https://icalendar.readthedocs.io/en/stable/how-to/usage.html#property-parameters and https://icalendar.readthedocs.io/en/stable/how-to/usage.html#add-timezones-to-a-calendar.
- [ ] How to create a freebusy request
- [ ] How to create a recurring event. Reuse Examples section in [vRecur](https://icalendar.readthedocs.io/en/latest/reference/api/icalendar.prop.recur.recur.html#icalendar.prop.recur.recur.vRecur) and from https://github.com/collective/icalendar/issues/421#issuecomment-1259369246.
- [ ] How to modify an event in a series
- [ ] How to remove an event that happens at a specific time
- [ ] How to remove all events in a series
- [ ] How to add an alarm to an event #1421
- [ ] How to make sure that timezone components are automatically added.
- [ ] add() is mentioned but not called
- [x] change pytz occurrences after #609 to zoneinfo
- [ ] Example heading should be more descriptive of what it actually does
- [ ] Overview of components
- [ ] Overview of properties
- [ ] File Structure could be moved more to the bottom because people might not be interested in it at the start but what to USE the module which is actually the name of the file - maybe it should be moved somewhere else. 
- [x] https://github.com/collective/icalendar/pull/847#issuecomment-2890613353 name responsibilities more clearly
- [ ] get attendees from an event, set responses as `PARSTAT` (participant status) for each attendee, and set set the attendees in the event. See https://github.com/collective/icalendar/issues/510.

See also:
- #125 
- #620

[usage.html]: https://icalendar.readthedocs.io/en/latest/usage.html

**Comment by stevepiercy** [comment]
@niccokunzmann I've added related issues as subissues to this primary issue for better tracking.

**Comment by stevepiercy** [comment]
@niccokunzmann while reviewing https://github.com/collective/icalendar/pull/1342, I realized that the examples use the Python dict method, not the modern icalendar API. I think that all the examples should be rewritten to use the icalendar API, and refer to https://icalendar.readthedocs.io/en/latest/explanation/api-design.html#property-access for an explanation of why `.add()` > `cal["prop"]`. What do you think?

**Comment by niccokunzmann** [comment]
It can be an issue to update to the new api.
We generally support them all. So, yes. It would be good to update them. Also, it is better to have docs than not to have them. So which API people use for creating docs should not hold contribution back. But the guidance is to use the attributes.

**Comment by stevepiercy** [comment]
I meant only to enhance the Usage page in this issue, not all the docs, which is why I wouldn't want to create a new issue. This page was written back in 2005 (perhaps earlier?), before the enhancements to the icalendar API. I think that not including, or not even mentioning, the icalendar API method is a disservice to the developer.

But taking the idea of supporting both APIs further, I think that it would be even better to use the tabbed interface, one for "icalendar API" and the other for "Python" (or perhaps you have better names?), with an example for each method. Readers could compare the differences, and make a better informed decision for choosing one method over another.

If this is something you think would be useful, then I could write the introduction and update an example with the tabbed interface that new contributors could use as a pattern. What do you think?

**Comment by jmes2build** [comment]
I'd like to take the **"How to save a file"** item, if it's still open.

Looking at `docs/how-to/usage.rst`, the neighbouring "Read calendar from file" section already reads cleanly via `pathlib`:

```pycon
>>> ics_path = Path("src/icalendar/tests/calendars/example.ics")
>>> calendar = icalendar.Calendar.from_ical(ics_path)
```

but the "Write to disk" example further down is still pre-`pathlib` and asymmetric with it:

```pycon
>>> import tempfile, os
>>> directory = tempfile.mkdtemp()
>>> f = open(os.path.join(directory, "example.ics"), "wb")
>>> f.write(cal.to_ical())
733
>>> f.close()
```

Three things I'd propose changing:

1. Use a context manager or `Path.write_bytes(cal.to_ical())` so the handle can't leak.
2. Drop the bare `733` expected output — it's a byte count that silently breaks the doctest any time the example calendar changes.
3. Mirror the `pathlib` style used by the read example just above, so read and write read as a pair.

Since `from_ical` accepts `str | bytes | Path` but there's no corresponding `to_file`/`save` helper, I'd also note explicitly in the prose that writing is the caller's responsibility — that asymmetry surprised me as a new reader.

Happy to open a single PR for just this item. Does that match what you had in mind, and is the `pathlib` style the direction you'd prefer?
